File: app/src/main/java/com/serwylo/lexica/db/Result.java
package com.serwylo.lexica.db;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Entity
@Data
@Builder
public class Result implements Parcelable {

    @PrimaryKey(autoGenerate = true)
    private long resultId;

    private long gameModeId;

    @NonNull
    private String langCode;

    private long score;

    private long maxScore;

    private int numWords;

    private int maxNumWords;

    protected Result(Parcel in) {
        resultId = in.readLong();
        gameModeId = in.readLong();
        langCode = in.readString();
        score = in.readLong();
        maxScore = in.readLong();
    }

    public static final Creator<Result> CREATOR = new Creator<Result>() {
        @Override
        public Result createFromParcel(Parcel in) {
            return new Result(in);
        }

        @Override
        public Result[] newArray(int size) {
            return new Result[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(resultId);
        dest.writeLong(gameModeId);
        dest.writeString(langCode);
        dest.writeLong(score);
        dest.writeLong(maxScore);
    }
}

File: app/src/main/java/com/serwylo/lexica/db/Result.kt
package com.serwylo.lexica.db

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize

/**
 * Each game that is played ends up recording a score in the database.
 * The score is recorded by this entity, on a per-game-mode-per-language basis (because different
 * languages may tend towards higher or lower scores due to dictionary differences).
 */
@Parcelize
@Entity
data class Result(
    @PrimaryKey(autoGenerate = true)
    val resultId: Long = 0,
    val gameModeId: Long,
    val langCode: String,
    val score: Long,
    val maxScore: Long,
    val numWords: Int = 0,
    val maxNumWords: Int = 0,
) : Parcelable